apiserver/facades/client/application/deploy.go

```diff
@@ -331,20 +317,6 @@ func (api *APIBase) addUnits(
 			return nil, internalerrors.Errorf("adding unit to application %q: %w", appName, err)
 		}
 		units = append(units, unitNames...)
-		if !assignUnits {
-			continue
-		}
-
-		// Are there still placement directives to use?
-		if i > len(placement)-1 {
-			if err := unit.AssignUnit(); err != nil {
-				return nil, internalerrors.Errorf("acquiring new machine to host unit: %w", err)
-			}
-		} else {
-			if err := unit.AssignWithPlacement(placement[i], allSpaces); err != nil {
-				return nil, internalerrors.Errorf("acquiring machine for placement %q to host unit: %w", placement[i], err)
```

Comment:
These can now also be cleaned up from the facade interfaces, and also from state, if you want

Reply:
Ah, good to know :+1: